### PR TITLE
Feat - Added profile page

### PR DESCRIPTION
**Pull request by SRUJANA-PENUGONDA13**
Added profile page with below features

- Add Profile Picture
- Bio
- Portfolio URL
- Follow/Unfollow

https://user-images.githubusercontent.com/72279128/178310131-1da5df84-df23-4f6e-9721-8777f93d610b.mp4

### CODE REVIEW

**Review by simran61 — APPROVED** [review]
Functionality working as expected. looks good

File: src/frontend/components/avatarWithName/AvatarWithName.js [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
+
   return (
     <React.Fragment>
-      <div className="avatarWithName-container flex-dir-row">
+      <div className="avatarWithName-container flex-dir-row cursor-pointer">
         <Avatar src="https://bit.ly/dan-abramov" />
-        <div className="post-user-details flex-dir-col">
+        <div
+          className="post-user-details flex-dir-col "
+          onClick={() => avatarHandler(console.log("User name: ", username))}
```
Comment (dev-enforced) [inline src/frontend/components/avatarWithName/AvatarWithName.js]: Didn't understand the parameter you passed inside the avatarHandler function !! As this function doesn't take any arguments as mentioned above. Could you clarify?

File: src/frontend/services/users/unfollowUser.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import axios from "axios";
+
+const unfollowUser = async (username) => {
+  const token = localStorage.getItem("token");
+  console.log("Unfollow User: ");
```
Comment (dev-enforced) [inline src/frontend/services/users/unfollowUser.js]: Can remove this comment

File: src/frontend/custom-hooks/OutsideClickHandler.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  const [resetMenu, setResetMenu] = useState(false);
+  useEffect(() => {
+    const handleClickOutside = (e) => {
+      if (ref.current && !ref.current.contains(e.target)) {
+        setResetMenu(true);
+      } else setResetMenu(false);
+    };
+    document.addEventListener("click", handleClickOutside, true);
+    return () => {
+      document.addEventListener("click", handleClickOutside, true);
```
Comment (dev-enforced) [inline src/frontend/custom-hooks/OutsideClickHandler.js:13]: Why are we repeating this line in both cases? Shouldn't it be document.removeEventListener("click",handleClickOutside,true) inside the return block

File: src/frontend/components/profileModal/ProfileModal.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  useEffect(() => {
+    if (profileModalType === "following") {
+      setUsers(user?.following || []);
+    } else if (profileModalType === "followers") {
+      setUsers(user?.followers || []);
+    }
+  }, [user]);
+
+  return (
+    <React.Fragment>
```
Comment (dev-enforced) [inline src/frontend/components/profileModal/ProfileModal.js:34]: Instead of mentioning React fragment Can simply write it as `
```
<> 
{JSX stuff}
</>
```

**Review by dev-enforced — APPROVED** [review]
The live preview is working as mentioned in the video . Had some minor queries which are mentioned below. Good work @SRUJANA-PENUGONDA13 !! Saw the mix of Chakra UI components and your custom made components as well so that was also good!! 
One issue which I felt I faced was that in the profile page of other users when I clicked on the list of users followed by them then in that list I was not able to navigate to any users page properly. 
Can check this out later on: https://deploy-preview-3--developers-hub.netlify.app/profile/ram@gmail.com
image
Whenever I click on Kartik it does not take me to Kartik's profile. Rest all is looking good. Can remove the console.logs along !!

Code base is also working fine

**Comment by poojakelkar** [comment]
Remove console logs
Add top margin to profile photo.
Other changes have already been mentioned.
otherwise Nice folder structure.

File: src/frontend/components/avatarWithName/AvatarWithName.js (hunk 1 above)
Comment (SRUJANA-PENUGONDA13) [reply src/frontend/components/avatarWithName/AvatarWithName.js]: It's is by mistake added. Removed

Comment (SRUJANA-PENUGONDA13) [reply src/frontend/components/avatarWithName/AvatarWithName.js]: Fixed the profile issue

**Pull request merged**